benchmark/functional/test_httpapi.py

```diff
@@ -0,0 +1,28 @@
+from testtools import TestCase
+
+from twisted.web.http import CREATED
+
+from ..httpapi import TxMongoBackend
+from ..test.test_httpapi import BenchmarkAPITestsMixin
+
+
+class TxMongoBenchmarkAPITests(BenchmarkAPITestsMixin, TestCase):
+    def setUp(self):
+        self.backend = TxMongoBackend()
+        self.addCleanup(self.backend.disconnect)
+        super(TxMongoBenchmarkAPITests, self).setUp()
+
+    def submit(self, result):
+        """
+        Submit a result.
+        """
+        req = super(TxMongoBenchmarkAPITests, self).submit(result)
+
+        def add_cleanup(response):
+            if response.code == CREATED:
+                location = response.headers.getRawHeaders(b'Location')[0]
+                self.addCleanup(lambda: self.agent.request("DELETE", location))
```

Comment:
:question: Did you consider a "DELETE all" function that could be done at the start
of each mongo test? The performance would be different, but it would likely
mean fewer cascading test failures if something goes wrong such that this
doesn't run.